### PR TITLE
refactor: replace char_u with char 12: remove `STRLEN` part 2

### PR DESCRIPTION
**Pull request by dundargoc**
Work on https://github.com/neovim/neovim/issues/459

### CODE REVIEW

**Comment by dundargoc** [comment]
Continuation of https://github.com/neovim/neovim/pull/20077

File: src/nvim/edit.c [hunk 1]
```diff
@@ -3186,12 +3186,12 @@ bool in_cinkeys(int keytyped, int when, bool line_is_empty)
           if (keytyped == (int)p[-1]
               || (icase && keytyped < 256
                   && TOLOWER_LOC(keytyped) == TOLOWER_LOC((int)p[-1]))) {
-            line = (char_u *)get_cursor_pos_ptr();
+            line = get_cursor_pos_ptr();
             assert(p >= look && (uintmax_t)(p - look) <= SIZE_MAX);
             if ((curwin->w_cursor.col == (colnr_T)(p - look)
                  || !vim_iswordc(line[-(p - look) - 1]))
```
Comment (zeertzjq) [inline src/nvim/edit.c:3192]:
```suggestion
                 || !vim_iswordc((uint8_t)line[-(p - look) - 1]))
```

File: src/nvim/edit.c [hunk 2]
```diff
@@ -3186,12 +3186,12 @@ bool in_cinkeys(int keytyped, int when, bool line_is_empty)
           if (keytyped == (int)p[-1]
```
Comment (zeertzjq) [inline src/nvim/edit.c:3186]:
```suggestion
          if (keytyped == (int)(uint8_t)p[-1]
```

**Review by zeertzjq — COMMENTED** [review]
Not sure if `in_cinkeys()` even works properly for multibyte, but still better to avoid changing behavior. Several comparison with chars in `look` may also need change.

**Comment by dundargoc** [comment]
@zeertzjq OK, I'll change `look` to `uint8_t`.

**Pull request merged**